test/iterators.jl

```diff
@@ -67,6 +67,13 @@ end
 @test length(zip(1:3,product(1:7,cycle(1:3)))) == 3
 @test length(zip(1:3,product(1:7,cycle(1:3)),8)) == 1
 
+# map
+# ----
+@testset "Iterators.map" begin
+    @test collect(Iterators.map(string, 1:3)::Base.Generator) == map(string, 1:3)
+    @test collect(Iterators.map(tuple, 1:3, 4:6)::Base.Generator) == map(tuple, 1:3, 4:6)
+end
```

Comment:
it might be good to test that `Iterators.map` can be applied to infinite iterables such as `Iterators.cycle(1:3)` and has the expected behaviour for the first n iterates.

Reply:
What is the reason why testing infinite iterators is interesting here?  I'm testing these two invocations since `Base.Generator` has different methods for them. So, I think it's somewhat meaningful to make sure we can hit these methods.  Other than that, I don't think we need to add tests specially for `Iterators.map` and they can be done in `Base.Generator`.  I think it makes sense to rename direct uses of `Base.Generator` in Base/stdlib.  But I think it'd be better to do this in a separate PR to make this PR more "atomic".